File: src/com/atslangplugin/ATSFileTypeDynamic.java
package com.atslangplugin;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

/**
 * Created by brandon on 12/16/14.
 */
public class ATSFileTypeDynamic extends LanguageFileType {
    public static final ATSFileTypeDynamic INSTANCE = new
            ATSFileTypeDynamic();

    private ATSFileTypeDynamic() {
        super(ATSLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "dats file";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Dynamic Applied Type System (ATS) language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "dats";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return ATSIcons.FILE;
    }
}

File: src/com/atslangplugin/ATSFileTypeDynamic.kt
package com.atslangplugin

import com.intellij.openapi.fileTypes.LanguageFileType

import javax.swing.*

/**
 * Created by brandon on 12/16/14.
 */
public class ATSFileTypeDynamic private() : LanguageFileType(ATSLanguage.INSTANCE) {

    override fun getName(): String {
        return "dats file"
    }

    override fun getDescription(): String {
        return "Dynamic Applied Type System (ATS) language file"
    }

    override fun getDefaultExtension(): String {
        return "dats"
    }

    override fun getIcon(): Icon? {
        return ATSIcons.FILE
    }

    companion object {
        public val INSTANCE: ATSFileTypeDynamic = ATSFileTypeDynamic()
    }
}
